packages/web/app/src/components/policy/form-helper.ts

```diff
@@ -0,0 +1,58 @@
+import { useFormikContext } from 'formik';
+import { RuleInstanceSeverityLevel } from '@/graphql';
+import type { PolicyFormValues } from './rules-configuration';
+
+export function useConfigurationHelper() {
+  const formik = useFormikContext<PolicyFormValues>();
+
+  return {
+    ruleConfig(id: string) {
+      return {
+        enabled: formik.values.rules[id]?.enabled ?? false,
+        severity: formik.values.rules[id]?.severity,
+        config: formik.values.rules[id]?.config,
+        getConfigAsString() {
+          return JSON.stringify(formik.values.rules[id]?.config, null, 2);
+        },
+        setConfig(property: string, value: any) {
+          const actualProp = property === '' ? '' : `.${property}`;
+
+          if (value && Array.isArray(value) && value.length === 0) {
+            formik.setFieldValue(`rules.${id}.config${actualProp}`, undefined, true);
+          } else {
+            formik.setFieldValue(`rules.${id}.config${actualProp}`, value, true);
+          }
+        },
+        getConfigValue<T>(property: string): T | undefined {
+          const levels = property.split('.');
+          let propName: string | undefined = undefined;
+          let obj = formik.values.rules[id]?.config;
+
+          do {
+            propName = levels.shift();
+
+            if (propName && propName !== '') {
```

Comment:
```suggestion
            if (propName) {
```
this case will be always `true`